mailpoet/assets/js/src/form_editor/store/controls.tsx

```diff
@@ -252,18 +268,17 @@ export const controls = {
   },
 
   STORE_LOCALLY(actionData) {
-    window.localStorage.setItem(
-      actionData.key,
-      JSON.stringify(actionData.value),
-    );
+    const { key, value } = actionData as Record<string, string>;
+    window.localStorage.setItem(key, JSON.stringify(value));
   },
 
   CALL_API,
 
   ENSURE_BROWSER_URL(actionData) {
+    const { formId } = actionData as Record<string, string>;
     let url = select('mailpoet-form-editor').getFormEditorUrl();
-    url = `${url}${actionData.formId}`;
-    if (window.location !== url) {
+    url = `${url}${formId}`;
+    if (window.location.href !== url) {
```

Comment:
Yes, that's a bug fix, added a separate commit for it :)

Reply:
Good point @rodrigoprimo, I had tested with the new change and couldn't  find anything breaking, basically the previous state would always run `history.replaceState` but since the url in some cases was the same nothing would get triggered, with the added change the `history.replaceState` is only invoked when the url changes which will trigger route listeners, I'd say this change is pretty safe, but if you have confidence that this change can be causing issues I'm happy to revert it back, ts-ignore and create a bug ticket, what do you think?

Reply:
Thanks for the additional details. In this case, we can proceed with things as they are.